src/applications/mhv-medications/containers/Prescriptions.jsx

```diff
@@ -563,47 +592,234 @@ const Prescriptions = () => {
+            data-testid="prescriptions-nav-link-to-refill"
+            data-dd-action-name={
+              dataDogActionNames.medicationsListPage.START_REFILL_REQUEST
+            }
+          >
+            Start a refill request
+          </Link>
+        </div>
+      </va-card>
+    );
+  };
+
+  const renderRefillAlert = () => {
+    if (!showRefillProgressContent) return null;
+
+    return (
+      <RefillAlert
+        dataDogActionName={
+          dataDogActionNames.medicationsListPage.REFILL_ALERT_LINK
+        }
+        activeRefills={activeRefills}
+      />
+    );
+  };
+
+  const renderMedicationsList = () => {
+    return (
+      <MedicationsList
+        pagination={pagination}
+        rxList={filteredList}
+        scrollLocation={scrollLocation}
+        selectedSortOption={selectedSortOption}
+        updateLoadingStatus={updateLoadingStatus}
+        onPageChange={handlePageChange}
+      />
+    );
+  };
+
+  const renderHeader = () => (
+    <>
+      <h1 data-testid="list-page-title" className="vads-u-margin-bottom--2">
+        Medications
+      </h1>
+      <p
+        className="vads-u-margin-top--0 vads-u-margin-bottom--4"
+        data-testid="Title-Notes"
+      >
+        {removeLandingPage ? (
+          <>
+            Bring your medications list to each appointment. And tell your
+            provider about any new allergies or reactions. If you use Meds by
+            Mail, you can also call your servicing center and ask them to update
+            your records.
+          </>
+        ) : (
+          <>
+            When you share your medications list with providers, make sure you
+            also tell them about your allergies and reactions to medications.
+          </>
+        )}
+        {!showAllergiesContent && (
+          <>
+            {' '}
+            If you print or download this list, we’ll include a list of your
+            allergies.
+          </>
+        )}
+      </p>
+      {showAllergiesContent && (
+        <a
+          href="/my-health/medical-records/allergies"
+          rel="noreferrer"
+          className="vads-u-display--block vads-u-margin-bottom--3"
+          data-testid="allergies-link"
+        >
+          Go to your allergies and reactions
+        </a>
+      )}
+    </>
+  );
+
+  const renderMedicationsContent = () => {
+    // No medications exist
+    const noMedications =
+      filteredList?.length === 0 &&
+      filterCount &&
+      Object.values(filterCount).every(value => value === 0);
+
+    // No medications match the current filter
+    const noFilterMatches =
+      filteredList?.length === 0 &&
+      filterCount &&
+      Object.values(filterCount).some(value => value !== 0);
+
+    // Medications exist and should be displayed
+    const hasMedications =
+      filteredList?.length > 0 || paginatedPrescriptionsList?.length > 0;
+
+    if (noMedications) {
+      return renderEmptyPrescriptions();
+    }
+
+    return (
```

Comment:
Large number of lines changed here but it amounts to a re-arrangement to make the logic a bit easier to follow.